### DISCUSSION TITLE
shadow registers

### DISCUSSION THREAD

**Discussion by gkamendje73** [rggen/rggen#225] [Q&A]
I was wondering if someone could give me some pointers on how to specify shadow registers in rggen.

**Comment by taichi-ishitani** [comment]
Hi @gkamendje73 ,

I think the `indirect` register type can be used for this purpose.

* Description
  * https://github.com/rggen/rggen/wiki/Register-Map-Specifications#indirect
* Example
  * https://github.com/rggen/rggen-sample/blob/92137aa2b90c145dfb59ef12c043603a8f101e41/block_0.yml#L123-L138

Please tell me more details of usage of the `shadow` register if the `indirect` register is not suitable.

**Comment by gkamendje73** [comment]
Hi @taichi-ishitani ,
Thanks for your answer. I will have a look at the indirect register type. 

I was looking for something similar to this https://github.com/lowRISC/opentitan/blob/master/util/reggen/README.md#shadow-registers

**Reply by taichi-ishitani** [reply]
It seems that RgGen's `indirect` register and OpenTitan's `shadow` register are different.
However, RgGen supports the plugin feature so you can create your own plugin and add such register type support.

**Reply by gkamendje73** [reply]
Thanks for confirming. 
I will be looking at the plugin interface.
